samples/net/openthread/coap_server/src/ot_coap_utils.c

```diff
@@ -0,0 +1,247 @@
+#include <logging/log.h>
+LOG_MODULE_REGISTER(ot_coap_utils, CONFIG_OT_COAP_UTILS_LOG_LEVEL);
+
+#include "ot_coap_utils.h"
+
+#include <net/net_pkt.h>
+#include <net/net_l2.h>
+#include <net/openthread.h>
+#include <openthread/message.h>
+#include <openthread/ip6.h>
+#include <openthread/coap.h>
+#include <openthread/thread.h>
+
+typedef struct {
+	struct otInstance *ot;
+	bool provisioning_enabled;
+	light_request_callback on_light_request;
+	provisioning_request_callback on_provisioning_request;
+} server_context_t;
+
+static server_context_t server_context = { .ot = NULL,
+					   .provisioning_enabled = false,
+					   .on_light_request = NULL,
+					   .on_provisioning_request = NULL };
+
+/**@brief Definition of CoAP resources for provisioning. */
+static otCoapResource provisioning_resource = { .mUriPath =
+							PROVISIONING_URI_PATH,
+						.mHandler = NULL,
+						.mContext = NULL,
+						.mNext = NULL };
+
+/**@brief Definition of CoAP resources for light. */
+static otCoapResource light_resource = { .mUriPath = LIGHT_URI_PATH,
+					 .mHandler = NULL,
+					 .mContext = NULL,
+					 .mNext = NULL };
+
+static struct otInstance *get_ot_instance(void)
+{
+	struct otInstance *instance = NULL;
+	struct net_if *iface;
+	struct openthread_context *ot_context;
+
+	iface = net_if_get_first_by_type(&NET_L2_GET_NAME(OPENTHREAD));
+	if (!iface) {
+		LOG_ERR("There is no net interface for OpenThread");
+		goto end;
+	}
+
+	ot_context = net_if_l2_data(iface);
```

Comment:
Is casting not needed here?

Reply:
> Is casting not needed here?

in C casting from void pointer is not needed